Source/DafnyCore/AST/TopLevelDeclarations.cs

```diff
@@ -328,6 +328,7 @@ abstract public class ModuleDecl : TopLevelDecl {
     return Signature;
   }
   public int Height;
+  public Token RootToken = new Token();
```

Comment:
A list is a special kind of a tree, isn't it?
I will not name it FirstToken because the first token is the first token being parsed, whereas this one is a "dummy" one to which the FirstToken will attach later.
 The metaphor is it's the root from which the FirstToken and any other tokens will grow on
I'm not sure about HeadToken. If they were presented to me before I had even designed this, I would still prefer RootToken.